src/RefractDataStructure.cc

```diff
@@ -10,8 +10,50 @@
 #include "RefractDataStructure.h"
 #include "refract/AppendDecorator.h"
 
+#include "RefractSourceMap.h"
+
 namespace drafter {
 
+    template <typename T, typename V = typename T::ValueType>
+    struct Append {
+        typedef T ElementType;
+        typedef V ValueType;
+        ElementType*& element;
+        Append(ElementType*& e) : element(e)
+        {
+        }
+
+        void operator()(const SectionInfo<ValueType>& value)
+        {
+            //throw std::logic_error("Can not append to primitive type");
+            // FIXME: snowcrash warn about "Primitive type can not have member"
+            // but in real it create "empty" member
+            //
+            // solution for now: set if element has no already value, otherwise silently ignore
+            if (element->empty()) {
+                element->set(value.section);
+                AttachSourceMap(element, value);
+            }
+        }
+    };
+
+    template <typename T>
+    struct Append<T, std::vector<refract::IElement*> > {
+        typedef T ElementType;
+        typedef typename T::ValueType ValueType;
```

Comment:
No this really does not work, if you remove `typename` compiler will return error:
```
error: missing 'typename' prior to dependent type name 'T::ValueType'
```

Please check things before you write them into comments.